creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -555,83 +406,35 @@ class SnapbackSM {
-      }
-
-      /**
-       * Select chunk of users to process in this run
-       *  - User is selected if (user_id % ModuloBase = currentModuloSlice)
-       *  - nodeUsers = array of objects of schema { primary, secondary1, secondary2, user_id, wallet }
-       */
-      nodeUsers = nodeUsers.filter(
-        nodeUser => (nodeUser.user_id % ModuloBase === this.currentModuloSlice)
-      )
-      decisionTree.push({
-        stage: '3/ select nodeUsers modulo slice',
-        vals: { nodeUsersSubsetLength: nodeUsers.length, moduloBase: ModuloBase, currentModuloSlice: this.currentModuloSlice },
-        time: Date.now()
-      })
-
-      // Compute content node peerset from nodeUsers (all nodes that are in a shared replica set with this node)
-      let peerSet = []
-      try {
-        peerSet = this.computeContentNodePeerSet(nodeUsers)
+      nodeUsers = await this.peerSetManager.getNodeUsers()
+      nodeUsers = this.peerSetManager.sliceUsers({ nodeUsers, moduloBase: ModuloBase, currentModuloSlice: this.currentModuloSlice })
 
-        decisionTree.push({
-          stage: '4/ computeContentNodePeerSet() Success',
-          vals: { peerSetLength: peerSet.size },
-          time: Date.now()
-        })
-      } catch (e) {
-        decisionTree.push({ stage: '4/ computeContentNodePeerSet() Error', vals: e.message, time: Date.now() })
-        throw new Error('processStateMachineOperation():computeContentNodePeerSet() Error')
-      }
-
-      /**
-       * Determine health for every peer & build list of unhealthy peers
-       * TODO change from sequential to chunked parallel
-       */
-      const unhealthyPeers = new Set()
-      try {
-        for await (const peer of peerSet) {
-          const peerIsHealthy = await this.determinePeerHealth(peer)
-          if (!peerIsHealthy) {
-            unhealthyPeers.add(peer)
-          }
-        }
+      decisionTree.push({ stage: 'getNodeUsers() and sliceUsers() Success', vals: { nodeUsersLength: nodeUsers.length }, time: Date.now() })
+    } catch (e) {
+      decisionTree.push({ stage: 'getNodeUsers() or sliceUsers() Error', vals: e.message, time: Date.now() })
+      throw new Error(`processStateMachineOperation():getNodeUsers()/sliceUsers() Error: ${e.toString()}`)
+    }
 
-        decisionTree.push({
-          stage: '5/ determinePeerHealth() Success',
-          vals: {
-            unhealthyPeerSetLength: unhealthyPeers.size,
-            healthyPeerSetLength: peerSet.size - unhealthyPeers.size,
-            unhealthyPeers: Array.from(unhealthyPeers)
-          },
-          time: Date.now()
-        })
-      } catch (e) {
-        decisionTree.push({ stage: '5/ determinePeerHealth() Error', vals: e.message, time: Date.now() })
-        throw new Error('processStateMachineOperation():determinePeerHealth() Error')
-      }
+    let unhealthyPeers = await this.peerSetManager.getUnhealthyPeers(nodeUsers)
```

Comment:
looks like you moved the `computePeerSet()` call into this `getUnhealthyPeers()` call - i can see why you did that, but i'm not convinced its better - i think keeping it as a discrete top level step is important. currently its not used anywhere outside the peerhealth calculation but it might be in the future

anyway, probably not worth changing for now. this definitely condenses the top level function a lot which is good

Reply:
yep thats fine, i can see why you did it this way. might need to pull this out later but fine for now